### DISCUSSION TITLE
can runtime models contain cycles

### DISCUSSION THREAD

**Discussion by varioustoxins** [pydantic/pydantic#4760] [Question] [answered]
Hi 

I have a model that reads in OK, but uses integer IDs on objects and lists of integers ids in fields to define cyclic relationships between objects. If I setup these cyclic relationships at runtime, I get errors in object equality comparisons because of infinite recursion in _iter and dict methods in BaseModel. I think this is because __eq__ in BaseModel does a complete decent of a branch of the model to check equality....

The objects I am modelling  have their own definition of equality as well, could I replace __eq__ with my own version or are there subtle traps here. Would I be better to look at _iter or is the whole idea of reading / creating things with pydantic which are not  DAGs  hopeless...

regards
Gary

**Comment by samuelcolvin** [comment, marked as answer]
Using you're own `__eq__` method should be fine.

We need to change how model equality works in V2, I believe someone has already taken a crack at a solution in a PR somewhere.

**Reply by varioustoxins** [reply]
thanks thats definitely reassuring, I hope the new version will cope with non DAG graphs!
